### PR TITLE
[nrf fromlist][nrfconnect] Improve CHIP_MEMORY_PROFILING config

### PR DESCRIPTION
**Pull request by ArekBalysNordic**
Enabled the full SHELL option when memory profiling is enabled.

### CODE REVIEW

File: config/nrfconnect/chip-module/Kconfig.features [hunk 1]
```diff
@@ -75,18 +75,31 @@ endif # CHIP_SPI_NOR
...
+	# Matter stack
 	select CHIP_STATISTICS
+	# Heap
 	select CHIP_MALLOC_SYS_HEAP_WATERMARKS_SUPPORT if CHIP_MALLOC_SYS_HEAP
-	select MBEDTLS_MEMORY_DEBUG if !CHIP_CRYPTO_PSA
-	select SYS_HEAP_RUNTIME_STATS if CHIP_MALLOC_SYS_HEAP
-	select KERNEL_SHELL
+	select SYS_HEAP_RUNTIME_STATS if CHIP_MALLOC_SYS_HEAP	
+	# Crypto
+	select MBEDTLS_MEMORY_DEBUG if CHIP_CRYPTO_MBEDTLS
```
Comment (Damian-Nordic) [inline config/nrfconnect/chip-module/Kconfig.features:84]: We don't define `CHIP_CRYPTO_MBEDTLS` Kconfig, do we?

File: config/nrfconnect/chip-module/Kconfig.features [hunk 2]
```diff
@@ -75,18 +75,31 @@ endif # CHIP_SPI_NOR
...
 	select OPENTHREAD_SHELL if !CHIP_WIFI
+	# Zephyr
+	select KERNEL_SHELL
 	help
 	  Enables features for tracking memory usage in Matter.
 
+# Enable full debug option when memory profiling is set.
+if CHIP_MEMORY_PROFILING
+
+config SHELL_MINIMAL
```
Comment (Damian-Nordic) [inline config/nrfconnect/chip-module/Kconfig.features:98]: We have a similar default in `Kconfig.defaults`. Which one is selected depends on whether a sample includes `Kconfig.defaults` or `Kconfig.features` first (and it seems we're not consistent wrt that). I suggest modifying `Kconfig.defaults` with proper:
```
default y if !CHIP_MEMORY_PROFILING
```
to not rely on the order of inclusion.

**Review by Damian-Nordic — APPROVED** [review]

**Comment by Damian-Nordic** [comment]
Also, can you please describe why SHELL_MINIMAL is undesired? It doesn't seem related to the profiling feature.

**Comment by ArekBalysNordic** [comment]
> Also, can you please describe why SHELL_MINIMAL is undesired? It doesn't seem related to the profiling feature.

It was requested by QA to have it because it helps with using all SHELL components like KERNEL_SHELL, CHIP_SHELL, and NET_SHELL.

File: config/nrfconnect/chip-module/Kconfig.defaults [hunk 3]
```diff
@@ -465,10 +465,10 @@ config SHELL_STACK_SIZE
     default 2616 if CHIP_WIFI
 
 config SHELL_MINIMAL
-    default y
+    default y if !CHIP_MEMORY_PROFILING
 
 config KERNEL_SHELL
-    default n
+    default n if !CHIP_MEMORY_PROFILING
```
Comment (LuDuda) [inline config/nrfconnect/chip-module/Kconfig.defaults:471]: Do we need it? Could we just enable KERNEL_SHELL inside CHIP_MEMORY_PROFILING Kconfig definition?

Comment (ArekBalysNordic) [reply config/nrfconnect/chip-module/Kconfig.defaults:471]: I did it to ensure that the order of kconfig includes doesn't matter as Damian noticed above.

Comment (ArekBalysNordic) [reply config/nrfconnect/chip-module/Kconfig.defaults:471]: So indeed, it is not needed then 😄 I thought that it also will be overwritten.

**Review by kkasperczyk-no — APPROVED** [review]

**Pull request merged**